static/app/views/replays/utils/useReplayEvent.tsx

```diff
@@ -176,14 +176,35 @@ function useReplayEvent({eventSlug, location, orgId}: Options): Result {
         );
       }
 
+      // Find LCP spans that have a valid replay node id, this will be used to
+      const highlights = mergedReplayEvent?.entries[0].data
+        .filter(({op, data}) => op === 'largest-contentful-paint' && data?.nodeId > 0)
+        .map(({start_timestamp, data: {nodeId}}) => ({
+          type: 6, // plugin type
+          data: {
+            nodeId,
+            text: 'LCP',
+          },
+          timestamp: Math.floor(start_timestamp * 1000),
+        }));
+
+      // TODO(replays): ideally this would happen on SDK, but due
+      // to how plugins work, we are unable to specify a timestamp for an event
+      // (rrweb applies it), so it's possible actual LCP timestamp does not
+      // match when the observer happens and we emit an rrweb event (will
+      // look into this)
+      const rrwebEventsWithHighlights = [...rrwebEvents, ...highlights].sort((a, b) => {
+        return a.timestamp - b.timestamp;
+      });
```

Comment:
Can we throw this into a method? otherwise i think it's easy for this try/catch to get huge.

Also, it's nicer that the rrweb and highlights are merged together. Easier to pass around 👍

Reply:
whether this makes it into a method now or not.. it's not the end of the world. There's plenty of reason to revisit the data that's being returned from here, so we can move it around at that time too.

Reply:
both moved here: https://github.com/getsentry/sentry/pull/34116/files#diff-e0d0c693da50c7052192b8fd082bb013735bffc5441335b1110ab30c7302a196R182-R190